### DISCUSSION TITLE
Where are all the relations? How to extract connection between ways and nodes?

### DISCUSSION THREAD

**Discussion by zfallahnejad** [osm2pgsql-dev/osm2pgsql#2189] [Q&A] [answered]
I used the osm2pgsql tool to insert the `openstreetmap` data for france into Postgres. I used their default.style as the style file. 
![sc12](https://github.com/osm2pgsql-dev/osm2pgsql/assets/5528548/496ecb46-9fe2-4dbb-b35c-b87298c8b444)

This tool created four tables:
* planet_osm_line: contains all imported ways
* planet_osm_point: contains all imported nodes with tags
* planet_osm_polygon: contains all imported polygons. 
* planet_osm_roads: contains a subset of planet_osm_line suitable for rendering at low zoom levels

I was wondering where are all the relations in these tables. It seems that there are other tables like `planet_osm_rels` which might include what I want but they are missing here. How can I have this table too? Is there another way to have relations in the tables too?

I also want to know how to extract the connection between ways and nodes. I don't see any connection between the ways in `planet_osm_polygon` table and the points in `planet_osm_point` table. How can I have this connection in the generated tables too?

**Comment by SomeoneElseOSM** [comment, marked as answer]
> I was wondering where are all the relations in these tables.

Relations with (multi)polygon geometry will be in planet_osm_polygon.

> I don't see any connection between the ways in planet_osm_polygon table and the points in planet_osm_point table. How can I have this connection in the generated tables too?

That rather depends on what you are trying to do.  osm2pgsql supports [more than one](https://osm2pgsql.org/doc/manual.html) output type, and another might be more appropriate, depending on what you are trying to do.  What columns you get, and whether something in the input is treated as an area or a closed line, is also under your control.  See e.g. [here](https://osm2pgsql.org/doc/manual.html#lua-tag-transformations).
